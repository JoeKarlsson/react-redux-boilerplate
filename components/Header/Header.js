import React from 'react';
import {
  Link,
} from 'react-router-dom';

class Header extends React.Component {
  render() {
    return (
      <div className="header">
        <h1>React Reddit</h1>
        <ul role="nav">
          <li><Link to="/">Home</Link></li>
          <li><Link to="/about">About</Link></li>
        </ul>
      </div>
    )
  }
};

export default Header;
