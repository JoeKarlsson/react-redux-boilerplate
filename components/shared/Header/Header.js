import React from 'react';
import {
  Link,
} from 'react-router-dom';
import styles from './Header.scss';


class Header extends React.Component {
  render() {
    return (
      <div className={styles.header}>
        <h1>React Reddit</h1>
        <ul role="nav">
          <li><Link to="/">Home</Link></li>
          <li><Link to="/about">About</Link></li>
        </ul>
      </div>
    )
  }
}

export default Header;
