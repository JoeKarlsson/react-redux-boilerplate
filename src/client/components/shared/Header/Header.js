import React from 'react';
import {
	Link,
} from 'react-router-dom';

const Header = () => {
	return (
		<div className="header">
			<h1>React Reddit</h1>
			<ul role="navigation">
				<li><Link to="/">Home</Link></li>
				<li><Link to="/about">About</Link></li>
			</ul>
		</div>
	);
};

export default Header;
