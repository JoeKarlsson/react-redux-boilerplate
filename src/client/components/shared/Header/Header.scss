$charade: #2a2937;

.header_bar {

	header {
		width: 100%;
		padding-top: 1%;
		padding-bottom: 1%;
		font-size: 1.25rem;
		display: flex;
		flex-direction: row;

		a {
			text-decoration: none;
		}

		ul {
			margin: 0;
			margin-bottom: 0;
			padding: 0;
			list-style: none;
			display: flex;
			flex-direction: row;

			li {
				padding: 0.5rem 1rem;
			}
		}

		.header_logo {
			padding: 0.5rem 1rem;
			padding-left: 5%;
			flex: 1 0 auto;
			order: 10;
		}

		.header_nav {
			padding-right: 5%;
			flex: 0 1 auto;
			order: 20;
		}
	}
}
