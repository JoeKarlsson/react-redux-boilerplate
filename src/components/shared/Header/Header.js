import React from 'react';
import {
	Link,
} from 'react-router-dom';
import styles from './Header.scss';

const Header = () => {
	return (
		<div className={styles.header}>
			<h1>React Reddit</h1>
			<ul role="navigation">
				<li><Link to="/">Home</Link></li>
				<li><Link to="/about">About</Link></li>
			</ul>
		</div>
	);
};

export default Header;
