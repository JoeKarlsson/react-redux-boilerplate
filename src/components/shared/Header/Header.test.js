import React from 'react';
import {
	shallow,
} from 'enzyme';
import renderer from 'react-test-renderer';
import {
	MemoryRouter,
} from 'react-router-dom';
import Header from './Header';

describe('Footer', () => {
	let wrapper;
	let inst;

	beforeEach(() => {
		wrapper = shallow(
			<MemoryRouter>
				<Header />
			</MemoryRouter>,
		);
		inst = wrapper.instance();
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered', () => {
				const component = renderer.create(
					<MemoryRouter>
						<Header />
					</MemoryRouter>,
				);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});

			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should be selectable by the class `header`', () => {
				expect(wrapper.dive().dive().is('.header')).toBe(true);
			});

			it('should mount in the full DOM', () => {
				expect(wrapper.dive().dive().find('.header').length).toBe(1);
			});

			it('should render to static HTML', () => {
				expect(wrapper.dive().dive().text()).toContain('React Reddit');
			});

			it('should have correct inital state', () => {
				const initialState = inst.state;
				const expectedIntialState = null;
				expect(initialState).toBe(expectedIntialState);
			});

			it('should not have any inital props', () => {
				const initialProps = inst.props;
				const expectedProps = {};
				expect(initialProps).toMatchObject(expectedProps);
			});
		});
	});
});
