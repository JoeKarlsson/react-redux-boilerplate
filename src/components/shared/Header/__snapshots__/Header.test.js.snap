// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is rendered 1`] = `
<div
  className="header"
>
  <h1>
    React Reddit
  </h1>
  <ul
    role="navigation"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </li>
  </ul>
</div>
`;
